File: 2028-find-missing-observations/2028-find-missing-observations.py
class Solution:
    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
        m = len(rolls)
        
        sum_m = sum(rolls)
        m_avg = sum_m/m
        remaining_sum = mean*(n + m) - sum_m
        
        max_sum = n * 6
        min_sum = n
        
        if remaining_sum > max_sum or remaining_sum < min_sum:
            return []
        
        distribute_mean = remaining_sum//n
        mod = remaining_sum % n
        
        new_list = [distribute_mean] * n
        for i in range(mod):
            new_list[i] += 1
        
        return new_list
            
        
        
            
            
        

File: 2028-find-missing-observations/2028-find-missing-observations.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int m = rolls.size();
        int sum = 0;
        for(int i = 0; i<m; i++){
            sum = rolls[i] + sum;
        }
        int remaining_sum = mean*(n + m) - sum;
        
        int max_sum = n * 6;
        int min_sum = n;
        
        if(remaining_sum > max_sum || remaining_sum < min_sum){
            return {}; 
        }
           
        
        int distribute_mean = int(remaining_sum/n);
        int mod = remaining_sum % n;
        
        vector<int> new_list(n, distribute_mean);
        for(int i = 0; i<mod; i++){
            new_list[i] += 1;
        }
            
        
        return new_list;
    }
};